File: internacao/src/main/java/br/com/basis/madre/service/projection/EventoCalendario.java
package br.com.basis.madre.service.projection;

import br.com.basis.madre.domain.Leito;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.ZonedDateTime;

@JsonInclude(JsonInclude.Include.NON_NULL)
public interface EventoCalendario {
    Long getId();

    Leito getLeito();

    ZonedDateTime getDataDoLancamento();

    @JsonProperty("title")
    String getTipoDoEventoNome();

    @JsonProperty("start")
    ZonedDateTime getDataInicio();

    @JsonProperty("end")
    ZonedDateTime getDataFim();

    interface TipoDoEvento {
        String getNome();
    }
}

File: frontend/src/app/internacao/models/evento-calendario.ts
import { ILeito, Leito } from './leito';

export interface IEventoCalendario {
    id?: number;
    dataDoLancamento?: Date;
    leito?: ILeito;
    title?: string;
    start?: Date;
    end?: Date;
}

export class EventoCalendario implements IEventoCalendario {
    public id: number;
    public dataDoLancamento: Date;
    public leito: Leito;
    public title: string;
    public start: Date;
    public end: Date;
}
